packages/@ourworldindata/components/src/GdocsUtils.ts

```diff
@@ -45,7 +45,10 @@ export function convertHeadingTextToId(headingText: Span[]): string {
     return urlSlug(spansToUnformattedPlainText(headingText))
 }
 
-function _getPrefixedPath(prefix: string, gdoc: OwidGdoc): string {
+function _getPrefixedPath(
+    prefix: string,
+    gdoc: { slug: string; content: { type?: OwidGdocType } }
```

Comment:
Can you explain this change? It seems fine because `undefined` is handled, but I'm just curious about where you encountered an undefined `content.type` 🙂

Reply:
sure! This comes from `Byline` not having access to the full gdoc, where I wanted to use `getCanonicalUrl()`.

`content.type` is possibly undefined in `OwidGdocPostContent` (which is why it is covered in the P.union below). It might be a chicken and egg thing, and we might not need the type to be possibly undefined in the first place, but I didn't look more into it.